index/scorch/snapshot_index.go

```diff
@@ -146,10 +148,18 @@ func (i *IndexSnapshot) newIndexSnapshotFieldDict(field string,
 	results := make(chan *asynchSegmentResult)
 	for index, segment := range i.segment {
 		go func(index int, segment *SegmentSnapshot) {
+			var prevBytesRead uint64
+			if seg, ok := segment.segment.(diskStatsReporter); ok {
```

Comment:
same interface checks were repeated at 152 and 159.
Can't we make that to 1?

Reply:
yeah done, my bad.